### PR TITLE
Migrate TextField to Material 3

### LINKED ISSUE #103537: Update text fields to support Material 3
As part of the [Material 3 specification](https://m3.material.io/components/text-fields/overview) there are updates to Text field's visuals:

Screen Shot 2022-05-11 at 1 56 04 PM

We need to update the `TextField` and possibly the `InputDecorator` widgets in the Flutter Material library to support this.

### PR DESCRIPTION
**Pull request by hannah-hyj**
fixed: #103537

Migrate TextField to Material 3.

The inputDecorator is already migrated  to M3 with fillColor, boderColor, iconColor, label text style, supporting text style.

So this PR is just to migrate  input text style because it's in TextField, not the InputDecorator.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/test/material/text_field_test.dart [hunk 1]
```diff
@@ -7107,6 +7107,57 @@ void main() {
...
+    expect(editableText.style.locale, themeData.textTheme.bodyLarge!.locale);
+    expect(editableText.style.wordSpacing, themeData.textTheme.bodyLarge!.wordSpacing);
+
+    // Properties set on TextStyle override theme
+    const Color setColor = Colors.red;
+    await tester.pumpWidget(buildFrame(const TextStyle(color: setColor)));
+    editableText = tester.widget(find.byType(EditableText));
+    expect(editableText.style.color, setColor);
+
+    // inherit: false causes nothing to be merged in from theme
```
Comment (HansMuller) [inline packages/flutter/test/material/text_field_test.dart:7151]: NICE

File: dev/tools/gen_defaults/lib/text_field_template.dart [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import 'template.dart';
+
+class TextFieldTemplate extends TokenTemplate {
+  const TextFieldTemplate(super.blockName, super.fileName, super.tokens);
+
+  @override
+  String generate() => '''
+// Generated version ${tokens["version"]}
+
+TextStyle _m3InputStyle(BuildContext context) => ${textStyle("md.comp.filled-text-field.label-text")}!;
```
Comment (HansMuller) [inline dev/tools/gen_defaults/lib/text_field_template.dart:14]: Filled text fields and outlined text fields use the same text style? Same color?

**Review by HansMuller — APPROVED** [review]
LGTM

File: dev/tools/gen_defaults/lib/text_field_template.dart (hunk 2 above)
Comment (hangyujin) [reply dev/tools/gen_defaults/lib/text_field_template.dart:14]: Yes they are the same

**Comment by amorenew** [comment]
@hangyujin  I tried the `useMaterial3: true` flag to enable the new Text Field but I can't see the new effect
Do I need to use the master channel?

Affected widgets: (not including TextField)
https://api.flutter.dev/flutter/material/ThemeData/useMaterial3.html

**Comment by nidegen** [comment]
How do I get the "filled" text field style in material 3?

Example on the left (Image from #103537)

**Pull request merged**